src/hotspot/share/code/codeBlob.cpp

```diff
@@ -751,3 +751,9 @@ void OptimizedEntryBlob::oops_do(OopClosure* f, const frame& frame) {
 JavaFrameAnchor* OptimizedEntryBlob::jfa_for_frame(const frame& frame) const {
   return &frame_data_for_frame(frame)->jfa;
 }
+
+void OptimizedEntryBlob::free(OptimizedEntryBlob* blob) {
+  assert(blob != nullptr, "caller must check for NULL");
+  JNIHandles::destroy_global(blob->receiver());
+  BufferBlob::free(blob);
+}
```

Comment:
We were freeing upcall stubs using CodeCache::free, but this doesn't track memory usage. I've added this method to properly handle freeing upcall stubs.